python-backend/app/api/mines/mine/resources/mine_map.py

```diff
@@ -0,0 +1,74 @@
+import uuid
+from datetime import datetime
+import json
+
+from flask import request, make_response, current_app
+from flask_restplus import Resource, reqparse, inputs, fields
+
+from app.api.mines.location.models.mine_map_view_location import MineMapViewLocation
+from app.extensions import api, cache, db
+from ....utils.access_decorators import requires_any_of, MINE_VIEW, MINESPACE_PROPONENT
+from ....utils.resources_mixins import UserMixin
+from ....constants import MINE_MAP_CACHE, TIMEOUT_12_HOURS
+
+# FIXME: Model import from outside of its namespace
+# This breaks micro-service architecture and is done
+# for search performance until search can be refactored
+
+mine_location = api.model('MineMapLocation', {
+    'latitude': fields.String,
+    'longitude': fields.String,
+})
+
+mine_map_list = api.model(
+    'MineMap', {
+        'mine_guid': fields.String,
+        'mine_name': fields.String,
+        'mine_no': fields.String,
+        'mine_location': fields.Nested(mine_location)
+    })
+
+
+class MineMapResource(Resource, UserMixin):
+    @api.doc(description='This endpoint returns a list of all mines to be displayed on the map.')
+    @api.response(
+        200,
+        'Returns a list of mines with less information to be displayed on the map.',
+        model=mine_map_list)
+    @requires_any_of([MINE_VIEW, MINESPACE_PROPONENT])
+    def get(self):
+        # Below caches the mine map response object in redis with a timeout.
+        # Generating and jsonifying the map data takes 4-7 seconds with 50,000 points,
+        # so caching seems justified.
+        #
+        # TODO: Use some custom representation of this data vs JSON. The
+        # json string is massive (with 50,000 points: 16mb uncompressed, 2.5mb compressed).
+        # A quick test using delimented data brings this down to ~1mb compressed.
+        map_result = cache.get(MINE_MAP_CACHE)
+        last_modified = cache.get(MINE_MAP_CACHE + '_LAST_MODIFIED')
+        if not map_result:
```

Comment:
Flagging this for later. I think this piece of the API knows far too much about how it's being consumed. It's location data, but there's nothing (from what I can tell) that forces it to be used by a map.

This should probably be more fully decoupled from the frontend use case at some point. Thoughts @Jsyro @nayr974 ?

Reply:
Primarily naming and comments at this point, correct?

Reply:
Yeah, I don't think it's any functional changes at this point, which is good -- but generalizing the naming could help us steer clear of adding map-specific functionality in the future.